Xamarin 4.8 ArgumentException: An item with the same key has already been added. Key: background

Hi,

I recently upgraded to Xamarin Forms 4.8 and now I see that whenever the application loads it throws the following exception: 

```csharp
Unhandled Exception:
System.TypeInitializationException: The type initializer for 'MagicGradients.GradientView' threw an exception. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.ArgumentException: An item with the same key has already been added. Key: background
  at System.Collections.Generic.Dictionary`2[TKey,TValue].TryInsert (TKey key, TValue value, System.Collections.Generic.InsertionBehavior behavior) [0x0015a] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/Dictionary.cs:572 
  at System.Collections.Generic.Dictionary`2[TKey,TValue].Add (TKey key, TValue value) [0x00000] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/Dictionary.cs:240 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:395 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00081] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:409 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/external/corefx/src/Common/src/CoreLib/System/Reflection/MethodBase.cs:53 
  at MagicGradients.StyleSheets.RegisterStyle (System.String name, System.Type targetType, System.String bindablePropertyName) [0x000a7] in <b9be2afa08cb4dab98018a1af7e6eeba>:0 
  at MagicGradients.GradientView..cctor () [0x00035] in <b9be2afa08cb4dab98018a1af7e6eeba>:0 
   --- End of inner exception stack trace ---
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.8(intptr,intptr,intptr)
  at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.8(intptr,intptr,intptr)
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'MagicGradients.GradientView' threw an exception. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.ArgumentException: An item with the same key has already been added. Key: background
  at System.Collections.Generic.Dictionary`2[TKey,TValue].TryInsert (TKey key, TValue value, System.Collections.Generic.InsertionBehavior behavior) [0x0015a] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/Dictionary.cs:572 
  at System.Collections.Generic.Dictionary`2[TKey,TValue].Add (TKey key, TValue value) [0x00000] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/Dictionary.cs:240 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:395 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00081] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:409 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/external/corefx/src/Common/src/CoreLib/System/Reflection/MethodBase.cs:53 
  at MagicGradients.StyleSheets.RegisterStyle (System.String name, System.Type targetType, System.String bindablePropertyName) [0x000a7] in <b9be2afa08cb4dab98018a1af7e6eeba>:0 
  at MagicGradients.GradientView..cctor () [0x00035] in <b9be2afa08cb4dab98018a1af7e6eeba>:0 
   --- End of inner exception stack trace ---
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.8(intptr,intptr,intptr)
  at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.8(intptr,intptr,intptr)
```
If I migrate back to V4.7 the application works as expected. 

I have also attached a sample application, which is using Xamarin Forms 4.8, and shows the exception.
 
[Xamarin Forms 4.8 Sample.zip](https://github.com/mgierlasinski/MagicGradients/files/5048747/Xamarin.Forms.4.8.Sample.zip)

--- a/MagicGradients/StyleSheets.cs
+++ b/MagicGradients/StyleSheets.cs
@@ -31,7 +31,7 @@ namespace MagicGradients
                 var itemProperty = stylePropertiesType.GetProperty("Item");
                 var attributes = itemProperty?.GetValue(styleProperties, new object[] { name });
 
-                attributes?.GetType().GetMethod("Add")?.Invoke(attributes, new[] { styleAttributeInstance });
+                attributes?.GetType().GetMethod("Insert")?.Invoke(attributes, new[] { 0, styleAttributeInstance });
             }
             else
             {

